pace-util/pace/util/grid/generation.py

```diff
@@ -2167,6 +2195,20 @@ def _calculate_edge_a2c_vect_factors(self):
         )
         return edge_vect_w, edge_vect_e, edge_vect_s, edge_vect_n
 
+    def _calculate_2d_edge_a2c_vect_factors(self):
+        edge_vect_e_2d = self.quantity_factory.zeros([util.X_DIM, util.Y_DIM], "")
+        edge_vect_w_2d = self.quantity_factory.zeros([util.X_DIM, util.Y_DIM], "")
+        shape = self.lon.data.shape
+        east_edge_data = self.edge_vect_e.data[self._np.newaxis, ...]
+        east_edge_data = self._np.repeat(east_edge_data, shape[0], axis=0)
+        west_edge_data = self.edge_vect_w.data[self._np.newaxis, ...]
+        west_edge_data = self._np.repeat(west_edge_data, shape[0], axis=0)
+        edge_vect_e_2d.data[:-1, :-1], edge_vect_w_2d.data[:-1, :-1] = (
+            east_edge_data[:-1, :-1],
+            west_edge_data[:-1, :-1],
+        )
```

Comment:
this initially made me worried, assigning values to existing quantities with the gpu backend has been finnicky in various situations, and this code won't get exercised on the gpu at the moment. But I see other methods here using this pattern on code that is used in the dycore and thus is tested on GPU, so looks good.